### ISSUE TITLE
"An established connection was aborted" error when running the tests on a Windows machine

### ISSUE THREAD

**Issue by AvivC** [Lukasa/hyper#216]
Hello,

I have forked and cloned the project. Running the tests using `py.test` on my Windows machine, two tests fail in `test_integration.py`: `test_connection_string` and `test_initial_settings`. Both fail with the following exception:

    error: [Errno 10053] An established connection was aborted by the software in your host machine

This exception is raised when calling `self._sslobj.read(len, buffer)` in `ssl.SSLSocket.read()`.

Any ideas how to solve this issue?
Thank you :)

Also, as an additional unrelated question: in order to contribute to the project, should I `pip install --editable` in the root folder?

**Comment by Lukasa** [comment]
Hey, thanks for taking a look at the project! The easiest way to run the tests is to use tox, which is what I recommend doing (it'll handle doing the installation).

The error message there is very unclear: is it consistent? That is, if you rerun the tests a few time, does it always appear and always appear on only those tests?

**Comment by AvivC** [comment]
Yes, the error always appears (also after restarting my machine), and always with the same two tests. I'm running Python 2.7.
BTW, in order to use tox, do I need to install Python 3.5, 3.4 and PyPy manually?

**Comment by Lukasa** [comment]
There is no requirement that you run tox for all the Python versions on your machine: just ignore the complaints about having them missing.

That's very interesting! Does it happen if you run *just those* tests (using the -k argument to pytest)?

**Comment by AvivC** [comment]
Yes, running:

    py.test -k "test_integration and test_connection_string or test_initial_settings" -vv

shows only these two tests failing.

I'll be happy to investigate this further later this day. Will keep you posted :)

**Comment by Lukasa** [comment]
that would be great. =) I'd love to know what's going on there.

**Comment by tbetbetbe** [comment]
This might be addressed by #221.  That change solves a similar issue to the one you have described that  I was seeing when testing on a Mac.

Can you pull that change and let me know if it resolved the issue ?
